src/polynomial/grouped_poly.rs

```diff
@@ -43,6 +45,35 @@ impl<F: PrimeField> From<HashMap<usize, Expression<F>>> for GroupedPoly<F> {
 }
 
 impl<F: PrimeField> GroupedPoly<F> {
+    fn new(expr: Expression<F>, num_of_poly: usize, num_of_challenge: usize) -> Self {
```

Comment:
comments needed for `fn new`, including explanation for `num_of_poly` and `num_of_challenge`.